### ISSUE TITLE
[docs] Show how to enable Kerberos in various runtimes

### ISSUE THREAD

**Issue by webknjaz** [qutebrowser/qutebrowser#8313]
With https://github.com/qutebrowser/qutebrowser/issues/4595#issuecomment-2369165892, I realized that different platforms have transitive dependencies pre-compiled differently. So I figured it'd be nice to document this.

What are the best pages in the docs to put this info in? FAQ? Configuring? Install? All 3?

It'd be good to document the following:
* the need to enable the `kerberos` USE-flag on Gentoo
* the need to make sure qtwebengine got compiled with Kerberos in other environments (I have no idea how)
* a working example of the args in `config.py`:
  ```python
  $ cat ~/.config/qutebrowser/config.py
  c.qt.args = [
    "auth-server-allowlist=*.fedoraproject.org,*.work.corp.com",
    "auth-server-whitelist=*.fedoraproject.org,*.work.corp.com",
    "disable-auth-negotiate-cname-lookup=true",
  ]
  config.load_autoconfig()
  ```

**Comment by The-Compiler** [comment]
> * the need to enable the `kerberos` USE-flag on Gentoo (https://wiki.gentoo.org/wiki/Qutebrowser#Troubleshooting)

That seems like something for the install docs, as it's Gentoo-specific

> * the need to make sure qtwebengine got compiled with Kerberos in other environments (I have no idea how)

From a quick look at [Archlinux](https://gitlab.archlinux.org/archlinux/packaging/packages/qt6-webengine/-/blob/edc0422b4bf525b379c83397c115e9e10246bd9e/PKGBUILD#L101), [Debian](https://salsa.debian.org/qt-kde-team/qt6/qt6-webengine/-/blob/b625d94071b744a0a3207856530162e6aa229b8b/debian/rules#L67), [NixOS](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/qt-6/modules/qtwebengine.nix#L198) they all compile with Kerberos support already, and as a user you typically don't recompile things from source (especially not something as big as QtWebEngine). So nothing to do there.

> a working example of the args in `config.py`: [...]

If those are confirmed working, I'd rather turn them into qutebrowser settings. Relevant Qt commits:

- [Support --auth-server-whitelist (260597) · Gerrit Code Review](https://codereview.qt-project.org/c/qt/qtwebengine/+/260597) (added in 5.13, so present in all versions we support)
- [Adaptations for Chromium 86 (318584) · Gerrit Code Review](https://codereview.qt-project.org/c/qt/qtwebengine/+/318584) (changes `whitelist` to `allowlist`, we shouldn't pass both, but instead find the right one to pass based on Chromium version (we still support Chromium 83 via Qt 5.15.2, we'll probably drop that at some point, 
but until then it needs a quick version-specific adjustment)

What does the `disable-auth-negotiate-cname-lookup=true` thing do? From [what I can gather](https://issues.chromium.org/issues/41108393), that was removed from Chromium in 2015, so that seems like a cargo-cult leftover?

tl;dr: Do things work as expected when you only pass `auth-server-allowlist`? If so, let's just turn that into a setting instead of documenting things.

**Comment by webknjaz** [comment]
> Do things work as expected when you only pass `auth-server-allowlist`?

I think it worked when I tested it. I'll need to re-check.

> What does the `disable-auth-negotiate-cname-lookup=true` thing do?

Not sure, but this was documented as needed in the internal docs. I'll need to dig deeper to understand. But I think it was supposed to prevent DNS lookups.


> From [what I can gather](https://issues.chromium.org/issues/41108393), that was removed from Chromium in 2015, so that seems like a cargo-cult leftover?

I think that thread is talking about removing the CLI argument and not the underlying setting.

One thing I like about configuring Chrome/Firefox is that they support policy files — they are JSON/YAML configs for some supported settings that are put into some system-wide location and picked up by the browsers automatically. On our company-provisioned systems, they just inject those and Kerberos is pre-configured, among other things. They can also list browser extensions to be auto-installed or forbid some toggles from being changed.

I suppose qutebrowser's `config.py` covers this partially, but it would be interesting to see it picking up the policy for Chrome or Chromium automatically (or with a one-line call in `config.py`).

> > * the need to make sure qtwebengine got compiled with Kerberos in other environments (I have no idea how)
> 
> From a quick look at [Archlinux](https://gitlab.archlinux.org/archlinux/packaging/packages/qt6-webengine/-/blob/edc0422b4bf525b379c83397c115e9e10246bd9e/PKGBUILD#L101), [Debian](https://salsa.debian.org/qt-kde-team/qt6/qt6-webengine/-/blob/b625d94071b744a0a3207856530162e6aa229b8b/debian/rules#L67), [NixOS](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/qt-6/modules/qtwebengine.nix?rgh-link-date=2024-09-24T09%3A52%3A32Z#L198) they all compile with Kerberos support already, and as a user you typically don't recompile things from source (especially not something as big as QtWebEngine). So nothing to do there.

I was thinking something along the lines of the “about” page of qutebrowser showing how the underlying libraries are compiled — I haven't been able to find this information in GUI, which would've been useful while debugging.

> > * the need to enable the `kerberos` USE-flag on Gentoo ([wiki.gentoo.org/wiki/Qutebrowser#Troubleshooting](https://wiki.gentoo.org/wiki/Qutebrowser#Troubleshooting))
> 
> That seems like something for the install docs, as it's Gentoo-specific

Makes sense. It already mentions some USE-flags and hints. And I've updated the doc over in the Gentoo wiki already.

**Comment by The-Compiler** [comment]
No API exists for those policies, see [[QTBUG-60778] Ability to set Chromium managed settings - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-60778). I also don't think info about how Qt was built exists at runtime.

**Comment by webknjaz** [comment]
Oh, too bad. It's sad that FF and Chrome have different policy formats. And Chrome vs. Chromium look into different locations. It would be easier to implement it if there was one standard schema...

**Comment by The-Compiler** [comment]
Important or not, passing `disable-auth-negotiate-cname-lookup=true` to `qt.args` will not do anything at all from what I can see.

**Comment by The-Compiler** [comment]
> Yeah, I guess I understand it better now. Still, wanted to document why it's there. Is there any other way to pass the setting to qtwebengine?

Only if Chromium provides a way to set such a "pref" thingy or policy settings via CLI arguments or env variables. From what I can see, there's [only a small list](https://chromium.googlesource.com/chromium/src/+/main/chrome/browser/prefs/chrome_command_line_pref_store.cc) that are exposed as CLI arguments, unfortunately.

**Comment by coiby** [comment]
Yeah, I can confirm `disable-auth-negotiate-cname-lookup=true` doesn't work. And as inspired by [@webknjaz's comment](https://github.com/qutebrowser/qutebrowser/issues/8313#issuecomment-2383090442),  to workaround this issue, you can add a DNS record to /etc/hosts manually.
